### ISSUE TITLE
Add a MultiplayerTimer node that can sync time along peers in a network

### ISSUE THREAD

**Issue by grazianobolla** [godotengine/godot-proposals#6104]
### Describe the project you are working on

I'm working on a multiplayer fast-paced game where I need to synchronize player states, along the many ways of doing so many industry standards include having a synced timer (to do things like Entity Interpolation, or Client Side Prediction etc etc).
This is a feature that EVERY single multiplayer game will have to implement at some point, and the code is most of the times very similar.

### Describe the problem or limitation you are having in your project

Having to write time synchronization code is not only cumbersome but could be easily avoided with this proposal, which would fit all multiplayer projects.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Having a MultiplayerTimer node will go along with the current MultiplayerSpawner and MultiplayerSynchronizer, players can add this node (might also be a singleton or similar) to a scene and access a time (be Unix time or time since server start) the important part is that this time is relative and different for each client, being slightly adjusted to compensate for latency, this way, different clients can agree on the same time (or with very little deviation from server time).

Extremely useful for many networking related things. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

```c#
void OnReceivePositionRpc(Vector3 position, float sentAt)
{
   var time = _multiplayerTimer.GetCurrentServerTime();
   float timeSinceSent = time - sentAt;
}
```

Without the MultiplayerTimer functionality, this code would not be possible, since we can't know for sure if the machine time is the same as the server time, might be delayed, might be 1 second behind, might be 5 seconds after, timeSinceSent would not be realiable

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, it requires some code to set up a synced timer.

### Is there a reason why this should be core and not an add-on in the asset library?

Every multiplayer game will eventually use the feature.

**Comment by Calinou** [comment]
cc @Faless

**Comment by Faless** [comment]
I never really needed to sync clocks since in my cases working with deltas was enough.
I think this can be easily implemented via a script, I made a "pinger" script some years ago which I attached to some other proposal (but never really published on the asset store IIRC) which could probably be adapted for the scope, I'll try to find it and post it below.

**Comment by Faless** [comment]
Ah, I actually did publish the 3.x version on github at least ( https://github.com/Faless/gd-pinger )

**Comment by dsnopek** [comment]
I agree that this should be left up to the game to handle. It's relatively simple to implement, but it requires bouncing a message from the server to the client and back. But you wouldn't really want Godot automatically sending these little message, you'd ideally want to mix this data in with other data that the client and server are already sending to each other (since it's just another few bytes), and that'd need to be done in a game-specific way.
